### ISSUE TITLE
Allow subfigures

### ISSUE THREAD

**Issue by dourouc05** [docbook/docbook#94]
Follow up of the mailing list thread https://lists.oasis-open.org/archives/docbook/201707/msg00000.html. 

Main issue: how to encode subfigures in DocBook? Something that looks like this (A and B being figures): 

```
    A                         B
    Fig. 1.1: Caption A       Fig. 1.2: Caption B
                Fig 1: Pictures
```

Many workarounds have been proposed in the aforementioned thread, but what should be the cleanest way to do so? My suggestion would be to allow `figure` within `figure` (the inner figures being subfigures; in this case, no `mediaobject` is allowed within the root `figure`, only for the subfigures). (If no new tag is required, then I guess TDG should be updated to reflect the decisions.)

About the formatting, the subfigures should be aligned in a row (as long as space allows, in which case another row can be added). 

The same question holds for tables and subtables (see https://tex.stackexchange.com/a/76543 for a use case, albeit with LaTeX rather than DocBook).

**Comment by LarryRowland** [comment]
Sorry it has taken so long to get back to you on this.

After discussing this issue and realizing that there are similar requirements for examples and other formal elements, the TC is proposing adding a `formalgroup ` element that would allow a set of children, all of the same type of formal element without introducing additional complexity in the schema for the formal elements to handle not permitting content at one level but permitting it at another level.  It also makes the intention of the structure clearer.  This would allow for something like:

```
  <formalgroup>
    <title>More Still</title>
    <figure>
      <title>Glamourous</title>
      <mediaobject>
        <textobject>
          <para>High</para>
        </textobject>
      </mediaobject>
    </figure>
    <figure>
      <title>Fast</title>
      <para>Hello</para>
    </figure>
  </formalgroup>
```
And also permit:
```
  <formalgroup align="center" columns="2" type="horiz">
    <title>Testing Formal Group</title>
    <equation>
      <title>Addition</title>
      <mathphrase>5 + 2</mathphrase>
    </equation>
    <equation>
      <title>Subtraction</title>
      <mathphrase>5 -2</mathphrase>
    </equation>
  </formalgroup>
```
and other similar constructs for other formal elements.  A type attribute similar to that used for simple lists would allow specifying horizontal or vertical and would combine with a column attribute to provide simple control of the representation with an align attribute to control simple alignment of the group on the page.  It would also be possible to specify a `floatstyle ` attribute to reference more complex CSS-bindings for the rendered elements.

**Comment by dourouc05** [comment]
That would be so much better than my proposal!

**Comment by kosek** [comment]
Implemented in 5.2b05
Schema is available for testing at https://docbook.org/xml/5.2b05/

Proposed attributes align, columns and type were underspecified. I think that layout requirements would be very varying for different users. Instead of these attributes I have added fgstyle attribute which can be used for specifying style for formalgroup -- stylesheets can then be customized to do the right think. Similarly to formal elements  there is floatstyle and pgwide attribute available.
